### ISSUE TITLE
Add support for signals and properties for non-introspected signals

### ISSUE THREAD

**Issue by wotzlaff** [romgrk/node-gtk#183]
The whole problem about non-introspected objected is keeping me busy. This is mainly due to the plugin-based nature of GStreamer where most objects are not introspected. (Or maybe I'm just not approaching the problem in the right way?)

I'd like to propose some improvements for non-introspected objects:
- There should be a way to set **properties**. Names and types should be available at runtime.
- There should be a way to connect to **signals**. Names are known at runtime and arguments (number and type) are handled in the `Marshal` function of the `Closure`.
- In GStreamer there's something called **named constants**. I'm not sure where they are coming from. But maybe these can also be provided at runtime.

I already tried to implement the handling of signals which we can discuss in a pull request.

**Comment by romgrk** [comment]
So signals & objects will have been dealt with soon, do you have more infor on named constants? A code example in any language would be fine.

**Comment by wotzlaff** [comment]
Sorry for the delay.

Maybe there is no such concept as named constants. The occur in the documentation of plugins in gstreamer. The names are based on the definition which is set up using `g_enum_register_static` (or `g_flags_register_static`). Then, the value corresponding to some particular name can be obtained by `g_enum_get_value_by_nick`. Probably it is possible to do the same in JS. But it is also possible to look up the values in the documentation and use them in the code. So there is no real need to introduce some special handling for named constants (in my opinion).

Basically this means that my third point in the issue can be omitted and the issue is solved.

**Comment by romgrk** [comment]
I see, I know that some C macros are not exported even though they contain important values, so it might be a similar issue here. Anyway, if you do find an example let me know. Thanks :)
